Submitted By:            Armin K. <krejzi at email dot com>
Date:                    2013-04-29
Initial Package Version: 5.17
Upstream Status:         unknown
Origin:                  Arch Linux (Giovanni Campagna)
Description:             Allow using obexd without systemd in the user session

Not all sessions run systemd --user (actually, the majority
doesn't), so the dbus daemon must be able to spawn obexd
directly, and to do so it needs the full path of the daemon.
---
 Makefile.obexd                      | 4 ++--
 obexd/src/org.bluez.obex.service    | 4 ----
 obexd/src/org.bluez.obex.service.in | 4 ++++
 3 files changed, 6 insertions(+), 6 deletions(-)
 delete mode 100644 obexd/src/org.bluez.obex.service
 create mode 100644 obexd/src/org.bluez.obex.service.in

--- a/Makefile.obexd
+++ b/Makefile.obexd
@@ -2,12 +2,12 @@
 if SYSTEMD
 systemduserunitdir = $(SYSTEMD_USERUNITDIR)
 systemduserunit_DATA = obexd/src/obex.service
+endif
 
 dbussessionbusdir = $(DBUS_SESSIONBUSDIR)
 dbussessionbus_DATA = obexd/src/org.bluez.obex.service
-endif
 
-EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service
+EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service.in
 
 if OBEX
 
--- a/obexd/src/org.bluez.obex.service
+++ /dev/null
@@ -1,4 +0,0 @@
-[D-BUS Service]
-Name=org.bluez.obex
-Exec=/bin/false
-SystemdService=dbus-org.bluez.obex.service
--- /dev/null
+++ b/obexd/src/org.bluez.obex.service.in
@@ -0,0 +1,4 @@
+[D-BUS Service]
+Name=org.bluez.obex
+Exec=@libexecdir@/obexd
+SystemdService=dbus-org.bluez.obex.service
